spring-cloud-gcp-bigquery/src/main/java/org/springframework/cloud/gcp/bigquery/integration/outbound/BigQueryFileMessageHandler.java

```diff
@@ -0,0 +1,279 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.bigquery.integration.outbound;
+
+import java.io.BufferedInputStream;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.channels.Channels;
+import java.time.Duration;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import com.google.cloud.bigquery.BigQuery;
+import com.google.cloud.bigquery.FormatOptions;
+import com.google.cloud.bigquery.Job;
+import com.google.cloud.bigquery.JobInfo.WriteDisposition;
+import com.google.cloud.bigquery.JobStatus.State;
+import com.google.cloud.bigquery.TableDataWriteChannel;
+import com.google.cloud.bigquery.TableId;
+import com.google.cloud.bigquery.WriteChannelConfiguration;
+
+import org.springframework.expression.EvaluationContext;
+import org.springframework.expression.Expression;
+import org.springframework.expression.common.LiteralExpression;
+import org.springframework.integration.expression.ExpressionUtils;
+import org.springframework.integration.handler.AbstractReplyProducingMessageHandler;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageHandlingException;
+import org.springframework.scheduling.TaskScheduler;
+import org.springframework.util.Assert;
+import org.springframework.util.StreamUtils;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.SettableListenableFuture;
+
+/**
+ * A {@link org.springframework.messaging.MessageHandler} which handles sending and
+ * loading files to a BigQuery table.
+ *
+ * @author Daniel Zou
+ * @since 1.2
+ */
+public class BigQueryFileMessageHandler extends AbstractReplyProducingMessageHandler {
+
+	private final BigQuery bigQuery;
+
+	private final TaskScheduler taskScheduler;
+
+	private final EvaluationContext evaluationContext;
+
+	private Expression datasetNameExpr;
+
+	private Expression tableNameExpr;
+
+	private FormatOptions formatOptions;
+
+	private boolean autoDetectSchema = true;
+
+	private WriteDisposition writeDisposition = WriteDisposition.WRITE_APPEND;
+
+	private Duration timeout = Duration.ofMinutes(5);
+
+	private boolean sync = false;
+
+	public BigQueryFileMessageHandler(BigQuery bigQuery, TaskScheduler taskScheduler) {
```

Comment:
There is ` getTaskScheduler()` and `setTaskScheduler()` in the super class.
So, this one must not be required from the ctor.

Reply:
Ok,  this is done, but I am afraid users will forget to do this so I added an assert as well:

```
Assert.notNull(getTaskScheduler(),
				"You must set a Task Scheduler for BigQueryFileMessageHandler before using "
						+ "by calling BigQueryFileMessageHandler.setTaskScheduler(taskScheduler).");
```

Reply:
Please, see that getter source code: it falls back into `IntegrationContextUtils.getTaskScheduler(this.beanFactory)` and that one is always there in the application context when we use Spring Integration, e.g. via `@EnableIntegration`.
Or Spring Boot auto-configuration like in most of our tests and samples...

Reply:
Ok, I updated the message so it mentions you must use Spring Integration or call setTaskScheduler; this will protect against users instantiating this class directly with `new BigQueryFileMessageHandler(..)`

I also updated the tests to autowire objects instead of use direct instantiation.